lib/jobs.js

```diff
@@ -37,6 +37,22 @@ const retryOptions = {
     }
 };
 
+/**
+ * Call executor.startPeriodic with the jobConfig obtained from the redis database
+ * @method startDelayed
+ * @param {Object}    jobConfig                 Configuration object
+ * @param {String}    jobConfig.jobId           Unique ID for a job
+ * @param {Function}  callback                  Callback fn(error, result)
+ */
+function startDelayed(jobConfig, callback) {
+    return redis.hget(`${queuePrefix}periodicBuilds`, jobConfig.jobId)
+        .then(fullConfig => executor.startPeriodic(JSON.parse(fullConfig), true))
```

Comment:
@pranavrc you can make a separate config section for the secrets to to construct the `executor-queue`. 

I don't think we should put it under the current `executor` (https://github.com/screwdriver-cd/queue-worker/blob/master/config/custom-environment-variables.yaml#L4-L6) because we will need to do extra logic in the `router`.

Reply:
If we don't want to instantiate a new `executor-queue` then we also just change the `executor-router` logic.